### PR TITLE
ANN: Detect E0049 (Implementation has wrong number of type or const parameters)

### PR DESCRIPTION
**Pull request by mchernyavsky**
Closes https://github.com/intellij-rust/intellij-rust/issues/8312

changelog: Detect [E0049](https://doc.rust-lang.org/error-index.html#E0049) (Implementation has wrong number of type or const parameters)

### CODE REVIEW

**Review by ozkriff — APPROVED** [review]

**Review by dima74 — COMMENTED** [review]
I am not sure that it is related to this PR, but there is a bug that error text can duplicate after editing type parameters:

Peek 2022-02-15 19-43

File: src/main/kotlin/org/rust/ide/inspections/RsWrongGenericParametersNumberInspection.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        getParameters: RsGenericDeclaration.() -> List<RsGenericParameter>
+    ) where T : RsAbstractable, T : RsGenericDeclaration {
+        val itemName = item.name ?: return
+        val itemType = when (item) {
+            is RsFunction -> "Method"
+            is RsTypeAlias -> "Type"
+            else -> return
+        }
+        val superItem = item.superItem as? RsGenericDeclaration ?: return
+        val typeParameterList = item.typeParameterList ?: return
```
[the remark this answers is not in the document]
Comment (mchernyavsky) [reply src/main/kotlin/org/rust/ide/inspections/RsWrongGenericParametersNumberInspection.kt]: Fixed.

**Comment by mchernyavsky** [comment]
@dima74 

> I am not sure that it is related to this PR, but there is a bug that error text can duplicate after editing type parameters:

The second message is from External Linter.

**Review by dima74 — APPROVED** [review]
bors r+

**Pull request merged**